Display current playlist

ethanberner opened the pull request:
Added the `DisplayCurrentPlaylist` component which allows users to see the current playlist. It displays time, song title, and artist. If there is no current playlist, the component renders empty. Rudimentary styling.

Component needs to be updated once we have the album art and a database.

Closes #18

Commit:
added song sample data

Commit:
new display component and setup

Commit:
added test

Commit:
updated test, updated playlist id in test-utls.js

Commit:
testing props

Commit:
Added basic playlist display functionality, needs more testing and styling

Commit:
changed date/time format to only show hour of time added

Commit:
updated description

Commit:
Rudimentary styling

Commit:
merged with main

Commit:
dealt with merge conflicts

Commit:
shows empty component when there is no current playlist

Commit:
add test

Commit:
fix broken test

ethansaxenian commented on src/components/DisplayCurrentPlaylist.js:
```diff
@@ -0,0 +1,40 @@
+
+const currentSongs = allSongs.filter((song) => song.playlistID === playlist.id);
+
+currentSongs.map((song) => song.timeAdded = new Date(song.timeAdded));
```
What is the purpose of this line? Remember that `map` returns a list, so you may want to use `forEach` here. But I'm not really sure what this code is trying to do.

madurst22 replied:
We were dealing with the standard `.format()` and converting it to 12 hour time. I believe `map` was necessary to use `.toLocaleString()`. Changed the `SongInput` to take the time using `.format("LT")` which gives us the time display we desired so I removed the `map`

ethansaxenian commented on src/components/DisplayCurrentPlaylist.js:
```diff
@@ -0,0 +1,40 @@
+
+currentSongs.map((song) => song.timeAdded = new Date(song.timeAdded));
+
+const currentPlaylist = currentSongs.map((song) => 
```
This looks like you're creating an array of unordered lists - I think these should be `<li>`s.

Also, this isn't a pressing issue but we may want to standardize how we deal with dates and just use the moment library for everything. Here is a cool [cheatsheet](https://devhints.io/moment) that I found. Perhaps we should store the song times in a different format to make this easier - we can talk about this as a group.

madurst22 replied:
I agree, made the change to `<li>`s. I would love to talk about dates and moment, I saw you're update in either `PlaylistDetails` or `ShowDetails` and had the same thought. That being said, the cheatsheet was what I needed. I changed the `SongInput` to the format type (`.format("LT")`)we were trying to achieve in the `DisplayCurrentPlaylist` component. We were trying to achieve standard 12 hour time (ex: 8:42 am), we can talk about if we want to display in a different format

ethansaxenian commented on src/pages/index.js:
```diff
@@ -95,10 +96,12 @@ export default function WRMCWebsite() {
               <p>{""}</p>
               <DisplayCurrentShow show={isOnAir ? upcomingShows[0] : shows.find(show => show.id === 12345)} handleClick={clickShow}/>
               <p>{""}</p>
+              <DisplayCurrentPlaylist playlist = {currentPlaylist ? currentPlaylist : []} allSongs = {allSongs}/>
```
This might cause a proptypes error, as `currentPlaylist` is an object, but sometimes we are passing in an empty list here.

madurst22 replied:
I did two things. First, I made it so that playlistType isn't required because an empty playlist may still sneak through if a show is started and a song isn't immediately entered. Second, I created an empty playlist object to be the false condition.

ethansaxenian reviewed (changes requested):
Well done! Just a few suggestions from me

Commit:
Pull Request Concerns addressed, empty array, unnecesary code, time format

ethansaxenian commented on src/components/DisplayCurrentPlaylist.js:
```diff
@@ -0,0 +1,38 @@
+        Current Playlist
+      </div>
+      <div className = {styles.playlist}>
+        {currentPlaylist}
```
Now I think we need to wrap `currentPlaylist` in `<ul>` tags.

ethansaxenian commented on src/pages/index.js:
```diff
@@ -95,10 +96,16 @@ export default function WRMCWebsite() {
+              <DisplayCurrentPlaylist playlist = {currentPlaylist ? currentPlaylist : 
+              {"date": "",
+              id: 12345,
+              showID: 12345}}
```
I don't think this actually works as an empty playlist, because it has an `id`, it will still display the songs with the same `playlistID`, if there are any. I think we need to simply pass `undefined` here and deal with it inside the `DisplayCurrentPlaylist` component.

ethanberner replied:
I'm wondering if that would still be in line with the discussion about ternary operators in class last week

ethansaxenian replied:
I believe that would be ok, as undefined is still a value. We could also pass in an empty object instead, but it might be harder to check in that case.

madurst22 replied:
I just made some changes. Removed the ternary operator and just passed through `currentPlaylist`. As Ethan S mentions, this will pass through undefined values at time. To deal with that I just added an if statement to the `DisplayCurrentPlaylist` component which checks to see if undefined was passed through

Commit:
fixed <ul> issue

Commit:
Resolve passing the empty playlist. Remove the ternary operator from index.js

ethansaxenian commented:
Looks great! (1/3)

schester77 reviewed (approved):
Looks good! Only thing I'd add is maybe we want to do something if there is no current playlist so it's not just an empty box that says 'current playlist'. Maybe have an else statement after the 'if(playlist !==undefined)' that adds some text saying 'tune in late' or something. 

(2/3)

Commit:
added message for when there is no show

ethanberner commented:
> Looks good! Only thing I'd add is maybe we want to do something if there is no current playlist so it's not just an empty box that says 'current playlist'. Maybe have an else statement after the 'if(playlist !==undefined)' that adds some text saying 'tune in late' or something.
> 
> (2/3)

Great point! I've addressed this in the latest commit and added a message for when there is no current playlist.

Cliffy2000 reviewed (approved):
looks great! 3/3

ethanberner committed:
Merge branch 'main' into display-current-playlist

Commit:
Fixed conflicts failing tests in index and DisplayCurrentPlaylist. Added cursor change over hover for the full schedule button.
